### ISSUE TITLE
Port pull #1311 to Tripal 4

### ISSUE THREAD

**Issue by dsenalik** [tripal/tripal#1483]
# Core Tripal 4 Development Task

## Pull #1484

Port changes from pull #1311 to Tripal 4 so that we can use in the GFF3 loader and others to replace the sql query for a sorted organism list (including by infraspecies), e.g. change in the gff3 loader:
```
<     $sql = "SELECT * FROM {organism} ORDER BY genus, species";
<     $org_rset = $chado->query($sql);
<     $organisms = [];
<     while ($organism = $org_rset->fetchObject()) {
<       $organisms[$organism->organism_id] = chado_get_organism_scientific_name($organism);
<     }
---
>     $organisms = chado_get_organism_select_options(FALSE, TRUE);

```

**Comment by dsenalik** [comment]
The first parameter to `chado_get_organism_select_options($syncd_only = TRUE, $show_common_name = FALSE)` is a relic from Tripal 2, and defaults to TRUE. Should we drop it? Would that break anything?
Too bad you guys didn't just use some `$args` back then 🤣

As I think on this, other modules may use it and get ported to Tripal 4 - I think it would be safest to leave it there as is, but trigger some sort of "no longer supported" error if left as TRUE.

**Comment by spficklin** [comment]
@laceysanderson can confirm, but it's my understanding we are not ensuring backwards compatibility with Tripal v2.  Is the function a relic of Tripal v2 or just the default setting?

**Comment by dsenalik** [comment]
@spficklin The parameter will be passed as TRUE only on a Tripal 2 site to select only "organisms sync'd with Drupal"
Nowhere in Tripal 3 or in any modules I have is it passed as TRUE.
If you try to pass TRUE on a Tripal 3 site you get an empty organism list ☠

**Comment by laceysanderson** [comment]
Yes, the $syncd_only is from T2 and I think it should be dropped entirely. The function is still very useful!

**Comment by dsenalik** [comment]
@laceysanderson do you mean remove it as a parameter? Or just disable as I have done.

**Comment by laceysanderson** [comment]
I was tempted to remove it as a parameter entirely... 🤔 

Although now that you pose the question perhaps we should just set it to FALSE by default and change the wording from `syncd` to `published`. Then later we could re-implement that parameter to only show organisms from chado which have been published within Tripal.

That way it's not a forever useless parameter but rather just a not yet implemented parameter 🤪

**Comment by dsenalik** [comment]
Okay, that is a nice solution, I'll do that, thanks.

**Comment by dsenalik** [comment]
Fixed with merged PR #1484
